### PR TITLE
Feat/group booking payment

### PR DESCRIPTION
**Pull request by silksil**
- Make payment of a group booking possible
- Make sure that a normal booking and group booking both make use of the `useCheckout` hook to to save global state in order to complete a group booking

### CODE REVIEW

File: src/pages/Checkout.tsx [hunk 1]
```diff
@@ -1,125 +1,77 @@
...
       pathname: '/bookings/confirmation',
       search: `?${createSearchParams({
-        offerId: checkout.offerId,
+        offerId: offerId || ' ',
         tx: result.tx.hash
       })}`,
-      hash: latestQueryParams?.location === 'Bogota' ? 'devcon' : ''
+      hash: bookingInfo?.location === 'Bogota' ? 'devcon' : ''
     });
   }, []);
```
Comment (bobetbat) [inline src/pages/Checkout.tsx:73]: missed dependencies bookingInfo, navigate

File: src/hooks/useCheckout/useCheckout.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    const { depositOptions, serviceId, providerId } = result;
+
+    // TODO: This needs to be revisited for the story to convert to different currencies
+    const quote: Quote = {
+      quoteId: 'dummy',
+      sourceCurrency: 'USD',
+      sourceAmount: depositOptions.usd || 'dummy',
+      targetCurrency: 'dummy',
+      targetAmount: 'dummy',
+      rate: 'dummy'
```
Comment (bobetbat) [inline src/hooks/useCheckout/useCheckout.ts:79]: possibly missed

File: src/hooks/useCheckout/useCheckout.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+   * and get necessary information to pay for the deposit.
+   */
+  const bookGroup = useMutation<GroupBookingRequestResponse, Error>(async () => {
+    if (!organizerInfo || !bookingInfo?.offers || !bookingInfo?.adultCount) {
+      throw new Error('Something went wrong. Please try selecting your rooms again.');
+    }
+    const { billingInfo, ...restOrganizerInfo } = organizerInfo;
+    const includeBillingInfo = bookingInfo?.invoice;
+
+    console.log(bookingInfo.invoice, billingInfo);
```
Comment (bobetbat) [inline src/hooks/useCheckout/useCheckout.ts:58]: lint

Comment (bobetbat) [reply src/hooks/useCheckout/useCheckout.ts:58]: rm logs

File: src/hooks/useCheckout/useCheckout.ts (hunk 2 above)
Comment (silksil) [reply src/hooks/useCheckout/useCheckout.ts:82]: discussed this with @blecorre and this is what he suggested for now. 

For the price filtering this is also gonna be refacotored on the BE, so then it is likely that we can remove it again.

Comment (silksil) [reply src/hooks/useCheckout/useCheckout.ts:82]: @blecorre let me know if i missed something

Comment (blecorre) [reply src/hooks/useCheckout/useCheckout.ts:82]: Well, just to be clear, I did not suggest you to write 'dummy' in it :) But indeed, sending the quote to the front-end adds unnecessary complexity, the logic source/target should stay in simard-pay. And the filtering functionality will add a second quote, bringing even more complexity. So I decided to abstract this complexity for group booking, and to limit the impact of this choice for now on the frontend, the best way was to workaround the Quote and generate a fake one on front end side. I hope we will have time to do the same in the standard flow when developing the filtering with the preferred currency flow so we can get rid of this quote.

**Review by bobetbat — APPROVED** [review]
LGTM

**Pull request merged**